### PR TITLE
Mediaviewer app

### PR DESCRIPTION
**Pull request by felixheidecke**
## Description

Adding media viewing capabilities. All still work in progress

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Open tasks:

- [x] Return to referrer on close
- [x] Show image name
- [x] Add download button

### CODE REVIEW

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUIFiles</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4121/3

<p>














</p></details>

File: apps/mediaviewer/src/Mediaviewer.vue [hunk 1]
```diff
@@ -0,0 +1,62 @@
+<template>
+  <div id="mediaviewer" class="uk-background-secondary">
+    <button @click="activeItem++">Next</button>
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/Mediaviewer.vue]: translate is missing

File: apps/mediaviewer/src/Mediaviewer.vue [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        <img v-show="!loading" ref="mediaviewerMedia" :src="image" alt="" style="max-width:90vw;max-height:90vh">
+      </transition>
+    </div>
+    <oc-spinner class="uk-position-center" v-if="loading" size="large" />
+  </div>
+</template>
+<script>
+import _ from 'lodash'
+import { mapGetters } from 'vuex'
+import imageSrc from '../../../src/mixins/imageSrc.js'
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/Mediaviewer.vue]: ouch - this is bad .... and should not be necessary - the directive is already registered in Vue

File: apps/mediaviewer/src/PdfViewerAppBar.vue [hunk 3]
```diff
@@ -0,0 +1,41 @@
+<template>
+  <div id="mediaviewer-app-bar" class="oc-app-bar">
+    <oc-grid flex gutter="small">
+      <div class="uk-width-expand">
+        <oc-icon name="application-pdf" /> Hi
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/PdfViewerAppBar.vue]: PDF :question:

File: apps/mediaviewer/src/app.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import '@babel/polyfill'
+// import translationsJson from '../l10n/translations'
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/app.js]: we need to bring this back - I'll take care in follow up PRs

File: apps/mediaviewer/src/Mediaviewer.vue (hunk 2 above)
Comment (felixheidecke) [reply apps/mediaviewer/src/Mediaviewer.vue]: In this case, it's a mixin. Not using the directive here as I needed the promise based attempt.

Comment (DeepDiver1975) [reply apps/mediaviewer/src/Mediaviewer.vue]: never the less - importing a file from outsode of the app can cause major issues in transpiling - been there - done that

File: apps/mediaviewer/src/Mediaviewer.vue [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    this.$forceUpdate()
+
+    // Return if no Image is selected
+    if (this.$store.getters.activeFile.path === '') {
+      this.closeApp()
+      return
+    }
+
+    // Set initial file
+    _.filter(this.mediaFiles, (file, index) => {
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/Mediaviewer.vue]: dont use underscore/lodash - use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

File: apps/mediaviewer/src/Mediaviewer.vue [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+      if (file.path === this.$store.getters.activeFile.path) {
+        this.activeIndex = index
+        console.log(index)
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/Mediaviewer.vue]: remove please

File: apps/mediaviewer/src/Mediaviewer.vue [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+      }
+    })
+  },
+  computed: {
+    ...mapGetters('Files', ['activeFiles']),
+    mediaFiles () {
+      return _.filter(this.activeFiles, file => {
+        return file.extension.toLowerCase().match(/(png|jpg|jpeg)/)
```
Comment (DeepDiver1975) [inline apps/mediaviewer/src/Mediaviewer.vue]: no gif? no raw image?

File: package.json [hunk 8]
```diff
@@ -111,5 +111,8 @@
       "eslint --fix",
       "git add"
     ]
+  },
+  "dependencies": {
+    "query-string": "^6.8.1"
```
Comment (DeepDiver1975) [inline package.json]: dev dependency

File: apps/mediaviewer/src/Mediaviewer.vue (hunk 2 above)
Comment (DeepDiver1975) [reply apps/mediaviewer/src/Mediaviewer.vue]: better use a plugin I guess - https://vuejs.org/v2/guide/plugins.html

**Comment by DeepDiver1975** [comment]
and please use the folder name 'media-viewer' THX

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUILogin</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4164/4

<p>





</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4163/9

<p>




</p></details>

File: package.json (hunk 8 above)
Comment (felixheidecke) [reply package.json]: [Done](https://github.com/owncloud/phoenix/pull/1530/commits/3327e225776179254c5d2f89f6d62493ec791255#diff-e208b8485ad370970bb4cdddc2e3e18fR31)

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4167/9

<p>
















</p></details>

File: apps/mediaviewer/src/Mediaviewer.vue (hunk 7 above)
Comment (felixheidecke) [reply apps/mediaviewer/src/Mediaviewer.vue]: Added gif, but my OC doesn't support any raw files :-/

**Comment by felixheidecke** [comment]
Will squash if all is good.

**Comment by felixheidecke** [comment]
Whilst working on the media viewer, I found out that data blobs don't get cached, so images may get fetched multiple times. I guess we could use the browsers local storage to do that, so I don't have to pollute the store with big data.

What do you think?

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4168/9

<p>




</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4168/13

<p>















</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4197/9

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong>webUISharingInternalUsers</strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4258/9

<p>









</p></details>

**Comment by ownclouders** [comment]
<details><summary>:boom: Acceptance tests <strong></strong> failed. Please find the screenshots inside ...</summary>

https://drone.owncloud.com/owncloud/phoenix/4258/12

<p>















</p></details>

**Comment by DeepDiver1975** [comment]
Squash please. As usual

File: apps/media-viewer/src/Mediaviewer.vue [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+    // Return if no Image is selected
+    if (this.$store.getters.activeFile.path === '') {
+      this.$router.push({
+        path: '/files'
+      })
+      return
+    }
+
+    // Set initial file
+    this.mediaFiles.filter((file, index) => {
```
Comment (DeepDiver1975) [inline apps/media-viewer/src/Mediaviewer.vue]: not really a filter - right? use foreach?

File: src/plugins/mediaSource.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+          fetch(source, { headers }).then(response => {
+            response.blob().then(blob => {
+              if (returnAs === 'base64') {
+                const reader = new FileReader()
+                reader.readAsDataURL(blob)
+                reader.onloadend = function () {
+                  resolve(reader)
+                }
+              } else if (returnAs === 'url') {
+                const source = window.URL.createObjectURL(blob)
```
Comment (DeepDiver1975) [inline src/plugins/mediaSource.js]: where is the corresponding delete?

**Pull request merged**